tests/conftest.py

```diff
@@ -209,17 +211,19 @@ def resetSnapshot(self):
         self.resetToSnapshot(self.captured)
 
     def createSnapshot(self):
-        self.chain.mine(1)
-        return {'block': self.chain.block, 'head_state': self.chain.head_state, 'contracts' : deepcopy(self.contracts), 'snapshot': self.chain.snapshot()}
+        return  { 'block': self.chain.block, 'head_state': self.chain.head_state, 'snapshot': self.chain.snapshot(), 'contracts': deepcopy(self.contracts) }
 
     def resetToSnapshot(self, captured):
         if len(captured) < 4:
             raise "captured snapshot doesn't have all parameters in dictionary, need to call createSnapshot"
         self.chain.block = captured['block']
-        self.chain.revert(captured['snapshot'])
         self.chain.head_state = captured['head_state']
+        self.chain.revert(captured['snapshot'])
         self.contracts = deepcopy(captured['contracts'])
 
+    def pushTimeStamp(self, setTime):
+        self.chain.head_state.timestamp = setTime
```

Comment:
What is the purpose of this function?  This changes call sites from an assignment to a function:
```
fixture.chain.head_state.timestamp += 10
```
```
fixture.pushTimestamp(fixture.chain.head_state.timestamp + 10)
```
The latter feels more awkward.

Reply:
an attempt to control chain changes, the timestamp changes added time to reporting window get end Time, dispute start time. They didn't refer to the chain's timestamp.